docs/source/examples/vqe-demo_custom-gate.myst

```diff
@@ -0,0 +1,282 @@
+---
+jupytext:
+  text_representation:
+    extension: .myst
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.10.3
+kernelspec:
+  display_name: Python 3 (ipykernel)
+  language: python
+  name: python3
+---
+
+# Variational Quantum Eigensolver improved with Mitiq
+In this example we investigate how Mitiq can help reduce errors
+and improve convergence of a simple VQE problem executed
+on a simulated noisy backend.
+
+
+The pyQuil VQE example shown here is adapted from
+https://grove-docs.readthedocs.io/en/latest/vqe.html#, 
+and the VQE function in Grove.
+
+```{code-cell} ipython3
+import numpy as np
+from pyquil import get_qc, Program
+from pyquil.gates import RX, RY, S, T, Z, CNOT, MEASURE
+from pyquil.paulis import PauliTerm, PauliSum, sZ
+from pyquil.noise import pauli_kraus_map, append_kraus_to_gate
+from typing import List, Union
+from collections import Counter
+from matplotlib import pyplot as plt
+from scipy import optimize
+import mitiq
+from mitiq import zne
+from mitiq.zne.scaling.folding import fold_gates_from_right
+```
+
+Use the get_qc command to initialize the simulated backend 
+where the pyQuil program will run
+
+```{code-cell} ipython3
+backend = get_qc("2q-qvm")
+```
+
+Example ansatz
+
+```{code-cell} ipython3
+program = Program()
+theta = program.declare("theta", memory_type="REAL")
+program += RX(theta, 0)
+program += T(0)
+program += CNOT(1, 0)
+program += S(0)
+program += Z(0)
+```
+
+Simulate depolarizing noise on the static gates
+
+```{code-cell} ipython3
+def add_noise_to_circuit(quil_prog):
+    """Define pyQuil gates with a custom noise model via Kraus operators:
+        1. Generate Kraus operators at given survival probability
+        2. Append Kraus operators to the gate matrices
+        3. Add custom gates to circuit
+    """
```

Comment:
Can you complete the docstring using Google style for input and output? More info and links [here in the documentation](https://mitiq.readthedocs.io/en/latest/contributing.html#style-guidelines).

Reply:
Done!